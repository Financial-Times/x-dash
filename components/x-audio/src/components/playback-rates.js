export const PLAYBACK_RATES = [1, 1.5, 0.5];

export const getNextPlaybackRate = currentRate => {
	const currentRateIndex = PLAYBACK_RATES.findIndex(value => value === currentRate);
	const nextRateIndex = currentRateIndex < PLAYBACK_RATES.length - 1 ? currentRateIndex + 1 : 0;

	return PLAYBACK_RATES[nextRateIndex];
}
