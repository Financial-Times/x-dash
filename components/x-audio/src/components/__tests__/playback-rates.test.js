import { getNextPlaybackRate, PLAYBACK_RATES } from '../playback-rates';

describe('playback-rates', () => {
	test('defaults to the first rate', () => {
		const nextPlaybackRate = getNextPlaybackRate();
		expect(nextPlaybackRate).toMatchSnapshot();
	});

	test('gets the next rate', () => {
		PLAYBACK_RATES.forEach(rate => {
			const nextPlaybackRate = getNextPlaybackRate(rate);
			expect(nextPlaybackRate).toMatchSnapshot();
		});
	});

	test('loops back to the first rate when at the end', () => {
		const nextPlaybackRate = getNextPlaybackRate(PLAYBACK_RATES[PLAYBACK_RATES.length - 1]);
		expect(nextPlaybackRate).toMatchSnapshot();
	});
});
