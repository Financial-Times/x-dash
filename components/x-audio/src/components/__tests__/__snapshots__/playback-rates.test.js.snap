// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playback-rates defaults to the first rate 1`] = `1`;

exports[`playback-rates gets the next rate 1`] = `1.5`;

exports[`playback-rates gets the next rate 2`] = `0.5`;

exports[`playback-rates gets the next rate 3`] = `1`;

exports[`playback-rates loops back to the first rate when at the end 1`] = `1`;
